lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1056,16 +1063,32 @@ estimateNumberOfWitnesses (Cardano.TxBody txbodycontent) =
             Cardano.TxCertificates _ certs _ -> length certs
             -- FIXME [ADP-1515] Not all certificates require witnesses. Will
             -- over-estimate unnecessarily.
+        txMintingScripts = length $ filter isScriptNative scripts
     in
     fromIntegral $
         length txInsUnique +
         length txExtraKeyWits' +
         length txWithdrawals' +
         txUpdateProposal' +
-        txCerts
+        txCerts +
+        txMintingScripts
```

Comment:
No this might have been a regression. 

This should be the old behaviour: https://github.com/input-output-hk/cardano-wallet/blob/2dfeac9e660f75be262877b77db5da041ad1aa4d/lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs#L1772-L1773 Which looks correct.